### ISSUE TITLE
Formulating a constraint with conditional

### ISSUE THREAD

**Issue by anthonyshook** [dirkschumacher/ompr#293]
Hello!  I'm wondering if the OMPR crowd can maybe help me think through the syntax for a particular constraint I'm trying to define.

I have this variable, which I've constrained to be the sum of a larger variable; after solving, it looks like this:
```
variable i value
1 groups 1     0
2 groups 2     0
3 groups 3     2
4 groups 4     2
```

Important context -- this is a prototypical problem, i.e., choose 4 from N, maximizing Y.  What this shows you is of those 4 that I selected, two came from group 3 and two from group 4.

The constraint I *want* to define, and am having trouble doing so here, is that I want the number of values in that variable that are *greater than 0* to be constrained to a specific number.  In other words, I want to be able to say "give me the optimal where at least N groups are represented"

As examples, let's say I added a `number of values >0 >=4`, then I should get:
```
variable i value
1 groups 1     1
2 groups 2     1
3 groups 3     1
4 groups 4     1
```
Whereas if I set it `number of values >0 >=3` I might get
```
variable i value
1 groups 1     0
2 groups 2     1
3 groups 3     1
4 groups 4     2
```
I've tried multiplying the value by it's own reciprocal, so that each number is transformed to 1, as in 
```add_constraint(sum_expr(groups[j] * 1/groups[j], j = 1:numGroups))``` 
but that gives me an error (which may or may not be related to a division by zero problem -- of course, `add_constraint(sum_expr(groups[j] * 1/groups[j] + 1, j = 1:numGroups))` gives me the same error)

That error is, of course 
```
Error in smallgroups[j] * 1/smallgroups[j] : 
  non-numeric argument to binary operator
```

I can't predefine what the values for multiplying the numbers and reducing them to 1 should be because they depend on the value of the variable itself; and I don't think I can pass the value of the variable to a function the way I would a masking / weight layer.

Any thoughts are appreciated!

**Comment by sbmack** [comment]
```
Error in smallgroups[j] * 1/smallgroups[j] : 
  non-numeric argument to binary operator
```
`smallgroups[j] * 1 / smallgroups[j]` is obviously equal to 1 for all j != 0.

Moreover 1 / variable makes a model non-linear.  There are ways to linearize depending on the intent.

**Comment by anthonyshook** [comment]
Thanks for the response!  I actually ended up doing one of those classic things where I banged my head against a problem for a day or two, came here to ask for help, then figured it out within the next hour.  Sometimes you just have to say the problem out loud...

Anyway, the solution was the big M solution.
For anyone else at this stage, I just made a second, binary variable, and put constraints on it like so:

```
Model %>%
  ompr::add_variable(smallgroups2[i], i = 1:4, type = 'binary') %>%
  ompr::add_constraint(smallgroups2[j] <= smallgroups[j], j = 1:4) %>%
  ompr::add_constraint(smallgroups[j] <= smallgroups2[j] * 10000, j=1:4)
```

From there, I can easily constrain that the sum_expr of smallgroups2 needs to be above a specific threshold, thereby constraining the model to require selection from N number of groups.
